pkgs/applications/networking/browsers/firefox/common.nix

```
@@ -489,6 +491,15 @@ buildStdenv.mkDerivation ({
     install -Dvm644 ${distributionIni} $out/lib/${binaryName}/distribution/distribution.ini
     install -Dvm644 ${defaultPrefsFile} $out/lib/${binaryName}/browser/defaults/preferences/nixos-default-prefs.js
 
+    # Install appstream metadata
+    mkdir -p $out/share/metainfo
+    # TODO: make input and output path configurable
+    substituteInPlace ../taskcluster/docker/firefox-flatpak/org.mozilla.firefox.appdata.xml.in \
+      --replace ' date="$DATE"' ""
+    VERSION="${version}" envsubst \
+      < ../taskcluster/docker/firefox-flatpak/org.mozilla.firefox.appdata.xml.in \
+      > $out/share/metainfo/org.mozilla.firefox.appdata.xml
```

Proposed argument names:
- appstreamMetadataLPath
- distributionId

```suggestion
    # Install appstream metadata
    mkdir -p $out/share/metainfo
    # TODO: make input and output path configurable
    substituteInPlace ../${appstreamMetadataSourcePath} \
      --replace ' date="$DATE"' ""
    VERSION="${version}" envsubst \
      < ../${appstreamMetadataSourcePath} \
      > $out/share/metainfo/${distributionId}.appdata.xml
```

---

reads fine to me